sklearn/callback/tests/_utils.py

```diff
@@ -50,10 +50,11 @@ def __init__(self, max_iter=20, computation_intensity=0.001):
         self.computation_intensity = computation_intensity
 
     @_fit_context(prefer_skip_nested_validation=False)
-    def fit(self, X=None, y=None):
-        callback_ctx = self.__skl_init_callback_context__(
-            max_subtasks=self.max_iter
-        ).eval_on_fit_begin(estimator=self)
+    def fit(self, X=None, y=None, X_val=None, y_val=None, callback_ctx=None):
+        if callback_ctx is not None:
+            callback_ctx.set_task_info(
+                task_name="fit", task_id=0, max_subtasks=self.max_iter
+            ).eval_on_fit_begin(estimator=self)
```

Comment:
`callback_ctx` can't be None

Reply:
The `@fit_callback` decorator will create the context and pass it no matter what, so the user has no way to prevent that. Anyway a user shouldn't deal with this param at all. And even if there's no callback attached to the estimator we need to build the tree of CallbackContexts because it may be a meta-estimator and its sub-estimator may have callbacks